bdk-ffi/Cargo.toml

```diff
@@ -17,12 +17,30 @@ path = "uniffi-bindgen.rs"
 default = ["uniffi/cli"]
 
 [dependencies]
-bdk = { version = "0.28.2", features = ["all-keys", "use-esplora-ureq", "sqlite-bundled", "rpc"] }
-uniffi = { version = "0.23.0" }
+bdk = { version = "1.0.0-alpha.2", features = ["all-keys", "keys-bip39"] }
+
+# TODO 22: The bdk_esplora crate uses esplora_client which uses reqwest for async. By default it uses the system
+#          openssl library, which is creating problems for cross-compilation. I'd rather use rustls, but it's hidden
+#          behind a feature flag. We need to look into whether openssl-sys is really required by bdk or if using rustls
+#          would work just as well. This here is a temporary workaround which removes the async feature on the bdk_esplora crate.
+#          See PR #1179 https://github.com/bitcoindevkit/bdk/pull/1179 for the fix in bdk.
+# bdk = { git = "https://github.com/thunderbiscuit/bdk.git", branch = "test-rust-tls", version = "1.0.0-alpha.2", features = ["all-keys", "keys-bip39"] }
+# bdk_esplora = { git = "https://github.com/thunderbiscuit/bdk.git", branch = "test-rust-tls", version = "0.4.0", package = "bdk_esplora", default-features = false, features = ["std", "blocking", "async-https-rustls"] }
+
+bdk_esplora = { version = "0.4.0", default-features = false, features = ["std", "blocking"] }
+uniffi = { version = "=0.23.0" }
 
 [build-dependencies]
-uniffi = { version = "0.23.0", features = ["build"] }
+uniffi = { version = "=0.23.0", features = ["build"] }
 
 [dev-dependencies]
-uniffi = { version = "0.23.0", features = ["bindgen-tests"] }
+uniffi = { version = "=0.23.0", features = ["bindgen-tests"] }
 assert_matches = "1.5.0"
+
+[profile.release-smaller]
+inherits = "release"
+opt-level = 'z'     # Optimize for size.
+lto = true          # Enable Link Time Optimization
+codegen-units = 1   # Reduce number of codegen units to increase optimizations.
+panic = "abort"     # Abort on panic
+strip = "debuginfo" # Partially strip symbols from binary
```

Comment:
I didn't have this change on my branch and also can't remember going over this, so it stood out to me that's all. I generally understand what this is trying to do but is this related to any specific Issue? Just for my own understanding or for reference is the only reason I'm asking.

Reply:
Here do you mean the release-smaller profile, or the change of the `strip = "debuginfo"` line?

The profile got moved from the root Cargo.toml because of the removal of the cargo workspace, but the strip = "debuginfo" is because our old strip = true prevents the `--library` flag from building the scaffolding code on Linux machines. This is explored more in thunderbiscuit/uniffi-examples#3, and it was timely that this guy happen to open the PR when he did because I ran into the issue just a few weeks later for this update.

But come to think of it this is not required _yet_, because we're not updating to the 0.24.3 version of uniffi-rs... I could revert that. No strong opinion on it yet; I've been meaning to test the difference in size for the binaries when the two flags are used.

Reply:
ahhh, that answers it. I was unsure if it was an all new block of code and a totally new addition to the project, but now that you mention it just got moved from a different file (the root `Cargo.toml`) because of the removal of the workspace that clears it up for me (not new code, but new spot!). Thanks!

Reply:
Also related is #405.